pkg/systemd/overview-cards/shutdownStatus.jsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useState, useEffect } from 'react';
+import { Button, Flex, FlexItem } from '@patternfly/react-core';
+import { PowerOffIcon, RedoIcon } from "@patternfly/react-icons";
+
+import * as timeformat from "timeformat";
+
+import cockpit from "cockpit";
+
+const _ = cockpit.gettext;
+const systemd_file = "/run/systemd/shutdown/scheduled";
+
+const getScheduledShutdown = (setShutdownTime, setShutdownType) => {
+    const client = cockpit.dbus("org.freedesktop.login1");
+    return client.call("/org/freedesktop/login1", "org.freedesktop.DBus.Properties", "Get", ["org.freedesktop.login1.Manager", "ScheduledShutdown"], { type: "ss" }).then(reply => {
+        const result = reply[0];
+        if (result) {
+            setShutdownType(result.v[0]);
+            setShutdownTime(result.v[1]);
+        }
+    });
+};
+
+const cancelShutdownAction = () => {
+    const client = cockpit.dbus("org.freedesktop.login1");
+    return client.call("/org/freedesktop/login1", "org.freedesktop.login1.Manager", "CancelScheduledShutdown").then(reply => {
+        if (!reply[0]) {
+            console.error("Unable to cancel shutdown");
+        }
+    });
+};
+
+export const ShutDownStatus = () => {
+    const [shutdownType, setShutdownType] = useState(null);
+    const [shutdownTime, setShutdownTime] = useState(0);
+
+    useEffect(() => {
+        if (shutdownType === null) {
+            getScheduledShutdown(setShutdownTime, setShutdownType);
+
+            // logind does not have a propertieschanged mechanism https://github.com/systemd/systemd/issues/22244
+            cockpit.file(systemd_file).watch(content => {
+                getScheduledShutdown(setShutdownTime, setShutdownType);
+            });
+        }
+    }, [shutdownType]);
```

Comment:
What happens if you change the scheduled time, but keep the type? Would that still treat this as a change and re-run the effect, even if it's getting set to the same value? It may be safer to not make that assumption, and re-run this on changes to `shutdownTime` as well.

Reply:
Good point that's wrong, so moved it out of the if statement.

Reply:
Still looks odd and asymmetric -- why not make the effect trigger list empty (so that it's effectively like `componentDidMount()`) and drop the `if`?

Reply:
Dropped the if, react requires an "dependency" to be passed so I've passed [] and this is only called once so it seems fine.

Reply:
Yes, `[]` is *significantly* different from `undefined` -- the former means "just run once", the latter "run on every render"